partner_communication_revision/data/demo.xml

```diff
@@ -0,0 +1,39 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <data>
+        <record id="template" model="mail.template">
+            <field name="name">Donation - Thank You Letter</field>
+            <field name="model_id" ref="partner_communication.model_partner_communication_job"/>
+            <field name="use_default_to" eval="True"/>
+            <field name="subject">Donation Receipt</field>
+            <field name="body_html" type="html">
+                <div>
+                    % set partner = object.get_objects()
+                </div>
+                <div>
+                    ${partner.salutation},
+                    <br/>
+                    <br/>
+                    We thank you for the donation of CHF .
+                    % if partner.partner:
```

Comment:
I guess you wanted to put `if partner.email` ? I think `partner` is not a valid field inside the `res.partner` object.